rero_ils/modules/stats/api.py

```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+# Copyright (C) 2020 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Stats for pricing records."""
+import arrow
+from dateutil.relativedelta import relativedelta
+from invenio_search.api import RecordsSearch
+
+from ..acq_order_lines.api import AcqOrderLinesSearch
+from ..documents.api import DocumentsSearch
+from ..items.api import ItemsSearch
+from ..libraries.api import LibrariesSearch
+from ..loans.logs.api import LoanOperationLog
+from ..patrons.api import PatronsSearch
+
+
+class StatsForPricing:
+    """Statistics for pricing."""
+
+    def __init__(self):
+        """Constructor."""
+        self.now = arrow.utcnow()
+        self.months_delta = 3
+        _to = self.now.format(fmt='YYYY-MM-DDT23:59:59')
+        _from = (self.now - relativedelta(
+            months=self.months_delta)).format(fmt='YYYY-MM-DDT00:00:00')
+        self.date_range = {'gte': _from, 'lte': _to}
+
+    def get_all_libraries(self):
+        """Get all libraries in the system."""
+        return LibrariesSearch().source(['pid', 'name', 'organisation']).scan()
+
+    def stats(self):
+        """Compute all the statistics."""
+        for lib in self.get_all_libraries():
+            print(f'==== {lib.name} (pid: {lib.pid}) =====')
```

Comment:
Do we use `click.echo` for printing?

Reply:
This will be removed in the next PR. We will use a REST API.